[DM-4788]: Added user docs and fragment lib update for the new provided option for files in software and firmware repository items. (#3472)
SoftwareAG/c8y-docs: content/device-integration/fragment-library-bundle/software.md

|----|----|----|
| name | Yes | The name of the software|
| version | Yes | A version identifier of the software|

> | url | No | A URL pointing to the location where the software file was obtained from. If the value is $PROVIDED, the device must resolve the binary itself|

Reviewer: $PROVIDED is only for operations, here we communicate the current status and url is already optional so I'd keep this as-is
```suggestion
| url | No | A URL pointing to the location where the software file was obtained from|
```

> | url | No | A URL pointing to the location where the software file was obtained from. If the value is $PROVIDED, the device must resolve the binary on its own|

Reviewer 2: ```suggestion
| url | No | A URL pointing to the location where the software file was obtained from. If the value is $PROVIDED, the device must resolve the binary itself|
```

| softwareType | No | An arbitrary string for organizing software artifacts|

The name and the version are used to identify the package. Already mentioned `c8y_SupportedSoftwareTypes` fragment
restricts possible software types that can be installed on the device.

> If the `url` is set to `$PROVIDED`, no file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. If the device cannot resolve the binary, the operation must be marked as **FAILED**.

Reviewer: Similarly to my comment of firmware, this should go into a section below Software Update as an additional explanation that in some cases URLs may contain $PROVIDED in both software list and software update operations. Also keep in mind that the target reader for this doc is a device integrator who in this case receives an operation that contains a URL with the value $PROVIDED and they need to understand how they are supposed to deal with that case in their device side implementation. I'd word it like this:

>  If the `url` field contains the value `$PROVIDED`, no file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. Typically this process works through package managers that retrieve the correct binary from the name, version and other device specific details. If the device cannot resolve the binary, the operation must be marked as **FAILED**.

**SmartREST example**

> For example, to mark a software as provided:

Reviewer 2: ```suggestion
To mark a software as provided:
```

5. Set operation status to SUCCESSFUL <br>
   `503,c8y_SoftwareUpdate`

{{< c8y-admon-info >}}

> If the URL is set to `$PROVIDED`, no file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. Typically, this process works through package managers that retrieve the correct binary from the name, version, and other device-specific details. If the device cannot resolve the binary, the operation is marked as FAILED.

Reviewer: ```suggestion
If the URL is set to `$PROVIDED`, no file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. Typically, this process works through package managers that retrieve the correct binary from the name, version, and other device-specific details. If the device cannot resolve the binary, it  must mark the operation as FAILED.
```

> If the url field contains the value $PROVIDED, no file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. Typically this process works through package managers that retrieve the correct binary from the name, version and other device specific details. If the device cannot resolve the binary, the operation must be marked as FAILED.

Reviewer 2: ```suggestion
If the URL is set to `$PROVIDED`, no file is uploaded or linked from the platform. In this case, the device is fully responsible for resolving the correct binary. Typically, this process works through package managers that retrieve the correct binary from the name, version, and other device-specific details. If the device cannot resolve the binary, the operation is marked as FAILED.
```
